File: native/sensor-bridge/ios/DragonflySensorBridge/Sources/DragonflySensorBridge/Adapters/DexcomAdapter.swift
import Foundation

/// Dexcom G6/G7 adapter. Real implementation will use the Dexcom Mobile
/// SDK (BLE) once the partner agreement is in place. Until then this
/// returns an empty discovery list and rejects start with `notImplemented`.
///
/// Vendor-specific quirks tracked here, none anywhere else:
/// - Dexcom session windows (~10 days G6 / ~10 days G7).
/// - 5-minute sample cadence.
/// - Backfill on reconnect: emit those readings with `readingKind = .backfill`.
public final class DexcomAdapter: SensorAdapter {
    public let vendor: GlucoseEvent.Vendor = .dexcom
    public private(set) var deviceName: String?
    public private(set) var isReady: Bool = false

    public init(deviceName: String? = "Dexcom") {
        self.deviceName = deviceName
        // Real impl flips isReady when the SDK has initialized.
    }

    public func discover() async throws -> [DiscoveredSensor] {
        // TODO(dexcom): scan via DexcomShareClient / Dexcom Mobile SDK.
        return []
    }

    public func pair(_ sensor: DiscoveredSensor) async throws {
        throw SensorAdapterError.notImplemented
    }

    public func start(onEvent: @escaping (GlucoseEvent) -> Void) async throws {
        throw SensorAdapterError.notImplemented
    }

    public func stop() async {
        // No-op until real BLE session exists.
    }
}

File: native/sensor-bridge/android/sensor-bridge/src/main/kotlin/com/dragonfly/sensorbridge/adapters/DexcomAdapter.kt
package com.dragonfly.sensorbridge.adapters

import com.dragonfly.sensorbridge.DiscoveredSensor
import com.dragonfly.sensorbridge.GlucoseEvent
import com.dragonfly.sensorbridge.SensorAdapter
import com.dragonfly.sensorbridge.SensorAdapterError

/**
 * Dexcom G6/G7 adapter. Real implementation will use the Dexcom Mobile
 * SDK (BLE) once the partner agreement is in place.
 *
 * Vendor quirks live here, none anywhere else:
 *  - 5-minute sample cadence
 *  - 10-day session windows; warmup before readings are valid
 *  - Backfill on reconnect → emit with readingKind = BACKFILL
 */
class DexcomAdapter(
    override val deviceName: String? = "Dexcom",
) : SensorAdapter {
    override val vendor: GlucoseEvent.Vendor = GlucoseEvent.Vendor.DEXCOM
    override val isReady: Boolean = false

    override suspend fun discover(): List<DiscoveredSensor> = emptyList()

    override suspend fun pair(sensor: DiscoveredSensor) {
        throw SensorAdapterError.NotImplemented
    }

    override suspend fun start(onEvent: (GlucoseEvent) -> Unit) {
        throw SensorAdapterError.NotImplemented
    }

    override suspend fun stop() {
        // No-op until real BLE session exists.
    }
}
